### ISSUE TITLE
SIGSEGV crash

### ISSUE THREAD

**Issue by naanlizard** [AirenSoft/OvenMediaEngine#847]
**Describe the bug**
Our OME instance crashed a week or so ago (and seems to keep crashing, from a quick look at our docker logs). SIGSEGV - unfortunately no other notable info I can tell

**Logs**
[omelog.txt](https://github.com/AirenSoft/OvenMediaEngine/files/9292907/omelog.txt)
[crash_20220803.dump.txt](https://github.com/AirenSoft/OvenMediaEngine/files/9292911/crash_20220803.dump.txt)

Dockerfile:

```
FROM airensoft/ovenmediaengine:0.14.5

RUN apt-get update && apt-get install -y curl

ENV API_ACCESS_TOKEN=x

WORKDIR /opt/ovenmediaengine/bin

COPY ./Server.xml /opt/ovenmediaengine/bin/origin_conf/Server.xml
COPY EdgeConfig.xml ./edge_conf/Server.xml

COPY entrypoint.sh .

HEALTHCHECK --interval=30s --timeout=30s --start-period=5s --retries=3 CMD curl --fail -H "Authorization: Basic $(printf ${API_ACCESS_TOKEN} | base64)"  'http://localhost:48081/v1/vhosts' 

CMD ["bash", "entrypoint.sh"]

```

Server.xml

```
<?xml version="1.0" encoding="UTF-8"?>

<Server version="8">
	<Name>OvenMediaEngine</Name>
	<!-- Host type (origin/edge) -->
	<Type>origin</Type>
	<!-- Specify IP address to bind (* means all IPs) -->
	<IP>*</IP>

	<!-- 
	To get the public IP address(mapped address of stun) of the local server. 
	This is useful when OME cannot obtain a public IP from an interface, such as AWS or docker environment. 
	If this is successful, you can use ${PublicIP} in your settings.
	-->
	<StunServer>stun.l.google.com:19302</StunServer>

	<!-- Settings for the ports to bind -->
	<Bind>

		<Managers>
			<API>
				<Port>48081</Port>
				<WorkerCount>1</WorkerCount>
			</API>
		</Managers>


		<Providers>
			<!-- Pull providers
			<RTSPC>
				<WorkerCount>1</WorkerCount>
			</RTSPC>
			<OVT>
				<WorkerCount>1</WorkerCount>
			</OVT>
			-->
			
			<!-- Push providers -->
			<RTMP>
				<Port>${env:OME_RTMP_PROV_PORT:1951}</Port>
				<WorkerCount>1</WorkerCount>
			</RTMP>
			<SRT>
				<Port>9999</Port>
				<WorkerCount>1</WorkerCount>
			</SRT>
			<WebRTC>
				<Signalling>
					<Port>${env:OME_SIGNALLING_PORT:3333}</Port>
					<WorkerCount>1</WorkerCount>
				</Signalling>

				<IceCandidates>

					<TcpRelay>*:3478</TcpRelay>
					<!-- TcpForce is an option to force the use of TCP rather than UDP in WebRTC streaming. (You can omit ?transport=tcp accordingly.) If <TcpRelay> is not set, playback may fail. -->
					<TcpForce>false</TcpForce>

					<IceCandidate>${env:OME_ICE_CANDIDATES:*:10000-10005/udp}</IceCandidate>
					<TcpRelayWorkerCount>1</TcpRelayWorkerCount>
				</IceCandidates>
			</WebRTC>
	
		</Providers>

		<Publishers>
			<WebRTC>
				<Signalling>
					<Port>${env:OME_SIGNALLING_PORT:3333}</Port>
					<!-- If you want to use TLS, specify the TLS port -->
					<TLSPort>${env:OME_SIGNALLING_PORT_SECURE:3334}</TLSPort>
					<WorkerCount>1</WorkerCount>
				</Signalling>
				<IceCandidates>
					<!-- 
						If you want to stream WebRTC over TCP, specify IP:Port for TURN server.
						This uses the TURN protocol, which delivers the stream from the built-in TURN server to the player's TURN client over TCP. 
						For detailed information, refer https://airensoft.gitbook.io/ovenmediaengine/streaming/webrtc-publishing#webrtc-over-tcp
					-->
					<TcpRelay>*:3478</TcpRelay>
					<!-- TcpForce is an option to force the use of TCP rather than UDP in WebRTC streaming. (You can omit ?transport=tcp accordingly.) If <TcpRelay> is not set, playback may fail. -->
					<TcpForce>false</TcpForce>

					<IceCandidate>${env:OME_ICE_CANDIDATES:*:10000-10005/udp}</IceCandidate>
					<TcpRelayWorkerCount>1</TcpRelayWorkerCount>
				</IceCandidates>
			</WebRTC>

			<HLS>
				<Port>799</Port>
				<!-- If you want to use TLS, specify the TLS port -->
				<TLSPort>800</TLSPort>
				<WorkerCount>1</WorkerCount>
			</HLS>

			<LLHLS>
				<Port>799</Port>
				<TLSPort>800</TLSPort>
				<WorkerCount>1</WorkerCount>
			</LLHLS>
		</Publishers>
	</Bind>

	<!-- P2P works only in WebRTC -->
	<!--
	<P2P>
		<MaxClientPeersPerHostPeer>2</MaxClientPeersPerHostPeer>
	</P2P>
	-->

	<!--
		Enable this configuration if you want to use API Server
		
		<AccessToken> is a token for authentication, and when you invoke the API, you must put "Basic base64encode(<AccessToken>)" in the "Authorization" header of HTTP request.
		For example, if you set <AccessToken> to "ome-access-token", you must set "Basic b21lLWFjY2Vzcy10b2tlbg==" in the "Authorization" header.
	-->

	<Managers>
		<Host>
			<Names>
				<Name>*.hostname.tld</Name>
				<Name>hostname.tld</Name>
				<Name>localhost</Name>
				<Name>127.0.0.1</Name>
			</Names>
			<TLS>
				<CertPath>/opt/ovenmediaengine/cert.pem</CertPath>
				<KeyPath>/opt/ovenmediaengine/privkey.pem</KeyPath>
				<ChainCertPath>/opt/ovenmediaengine/fullchain.pem</ChainCertPath>
			</TLS>
		</Host>
		<API>
			<AccessToken>${env:API_ACCESS_TOKEN:x}</AccessToken>
		</API>
	</Managers>


	<VirtualHosts>
		<!-- You can use wildcard like this to include multiple XMLs -->
		<VirtualHost>
			<Name>default</Name>
			<!--Distribution is a value that can be used when grouping the same vhost distributed across multiple servers. This value is output to the events log, so you can use it to aggregate statistics. -->
			<Distribution>hostname.tld</Distribution>

			<!-- Settings for multi ip/domain and TLS -->
			<Host>
				<Names>
						<Name>*.hostname.tld</Name>
						<Name>hostname.tld</Name>
						<Name>localhost</Name>
						<Name>127.0.0.1</Name>
				</Names>
				<TLS>
					<CertPath>/opt/ovenmediaengine/cert.pem</CertPath>
					<KeyPath>/opt/ovenmediaengine/privkey.pem</KeyPath>
					<ChainCertPath>/opt/ovenmediaengine/fullchain.pem</ChainCertPath>
				</TLS>
			</Host>
			
			<!-- Settings for applications -->
			<Applications>
				<Application>
					<Name>live</Name>
					<!-- Application type (live/vod) -->
					<Type>live</Type>
					<OutputProfiles>
						<!-- Enable this configuration if you want to hardware acceleration using GPU -->
						<HardwareAcceleration>false</HardwareAcceleration>
						<OutputProfile>
							<Name>bypass_stream</Name>
							<OutputStreamName>${OriginStreamName}</OutputStreamName>
							<Encodes>
								<Audio>
									<Bypass>true</Bypass>
								</Audio>
								<Video>
									<Bypass>true</Bypass>
								</Video>
								<Audio>
									<Codec>opus</Codec>
									<Bitrate>128000</Bitrate>
									<Samplerate>48000</Samplerate>
									<Channel>2</Channel>
								</Audio>
							</Encodes>
						</OutputProfile>

						<!-- Audio only -->
						<OutputProfile>
							<Name>audio_only</Name>
							<OutputStreamName>${OriginStreamName}_audio</OutputStreamName>
							<Encodes>
								<Audio>
									<Bypass>true</Bypass>
								</Audio>
								<Audio>
									<Codec>opus</Codec>
									<Bitrate>128000</Bitrate>
									<Samplerate>48000</Samplerate>
									<Channel>2</Channel>
								</Audio>
							</Encodes>
						</OutputProfile>

					</OutputProfiles>
					<Providers>
						<WebRTC />
						<RTMP />
						<WebRTC>
							<Timeout>30000</Timeout>
						</WebRTC>
					</Providers>
					<Publishers>
						<AppWorkerCount>1</AppWorkerCount>
						<StreamWorkerCount>8</StreamWorkerCount>

						<WebRTC>
							<Timeout>30000</Timeout>
							<Rtx>true</Rtx>
							<Ulpfec>true</Ulpfec>
							<JitterBuffer>true</JitterBuffer>
						</WebRTC>

						<RTMPPush></RTMPPush>

						<HLS>
              <SegmentDuration>1</SegmentDuration>
              <SegmentCount>3</SegmentCount>
              <CrossDomains>
                <Url>hostname.tld</Url>
              </CrossDomains>
            </HLS>

						<LLHLS>
							<ChunkDuration>0.2</ChunkDuration>
							<SegmentDuration>6</SegmentDuration>
							<SegmentCount>10</SegmentCount>
							<CrossDomains>
								<!-- <Url>hostname.tld</Url>
								<Url>*.hostname.tld</Url>
								<Url>https://staging.hostname.tld:8888</Url> -->
								<Url>*</Url>
							</CrossDomains>
						</LLHLS>
					</Publishers>
				</Application>
			</Applications>
		</VirtualHost>
	</VirtualHosts>
</Server>
					  
```

**Comment by getroot** [comment]
I took a hint from the crash dump you posted and patched the code that could possibly cause one SIGSEGV. 0.14.7 will be released soon, please test again with it. Thanks for reporting the bug.

**Comment by naanlizard** [comment]
Hello, still on the old version used here - another crash dump that we ran into.

Does it look like the same bug as above? If so, we can close this again, otherwise I will open a new ticket.

We'll be updating to the latest version shortly as well, been waiting on various bug to be fixed
